### PR TITLE
docs: Improve the Features section in README

### PR DESCRIPTION
**Pull request by vdusek**

### CODE REVIEW

**Review by honzajavorek — COMMENTED** [review]
I added two comments which I think improve the spelling or wording.

The rest is my subjective commentary on the matter of Crawlee and Scrapy comparison, which you can take as a feedback, but also completely ignore. I wanted to provide an outsider perspective, but at the same time, I think the framework creators should have the freedom to express their opinionated view and their ambition how the project should differentiate.

Just chiming in, not approving nor disapproving.

File: README.md [hunk 1]
```diff
@@ -185,10 +185,12 @@ Why Crawlee is the preferred choice for web scraping and crawling?
...
 
-- Crawlee has out-of-the-box support for **headless browser** crawling (Playwright).
-- Crawlee has a **minimalistic & elegant interface** - Set up your scraper with fewer than 10 lines of code.
-- Complete **type hint** coverage.
-- Based on standard **Asyncio**.
+- **Asyncio-based** – Leveraging the standard [Asyncio](https://docs.python.org/3/library/asyncio.html) library, Crawlee delivers better performance and seamless compatibility with other modern asynchronous libraries.
+- **Type hints** – Crawlee is a new project built with modern Python, and complete type hint coverage for a better developer experience.
+- **Minimalistic and elegant interface** – Set up a functional scraper in fewer than 10 lines of code, keeping things clean and simple.
+- **Headless browser support** – Crawlee natively supports the headless browser crawling via [Playwright](https://playwright.dev/python/).
+- **State persistence** – Unlike Scrapy, Crawlee supports state persistence during interruptions, saving time and costs by avoiding the need to restart scraping pipelines from scratch after an issue.
```
[the remark this answers is not in the document]
Comment (janbuchar) [reply README.md:192]: The Scrapy integration just uses the cloud storage when you run it on Apify, and that is persistent by design. But I don't think that most of our audience even knows that the Scrapy x Apify integration exists.

[the remark this answers is not in the document]
Comment (honzajavorek) [reply README.md:192]: I got carried away in the middle paragraph. It's more of an issue for the Scrapy template repo than something relevant here.

File: README.md [hunk 2]
```diff
@@ -185,10 +185,12 @@ Why Crawlee is the preferred choice for web scraping and crawling?
 
 ### Why to use Crawlee rather than Scrapy?
 
-- Crawlee has out-of-the-box support for **headless browser** crawling (Playwright).
-- Crawlee has a **minimalistic & elegant interface** - Set up your scraper with fewer than 10 lines of code.
-- Complete **type hint** coverage.
-- Based on standard **Asyncio**.
+- **Asyncio-based** – Leveraging the standard [Asyncio](https://docs.python.org/3/library/asyncio.html) library, Crawlee delivers better performance and seamless compatibility with other modern asynchronous libraries.
+- **Type hints** – Crawlee is a new project built with modern Python, and complete type hint coverage for a better developer experience.
+- **Minimalistic and elegant interface** – Set up a functional scraper in fewer than 10 lines of code, keeping things clean and simple.
```
[the remark this answers is not in the document]
Comment (honzajavorek) [reply README.md:190]: > I really like the fact that Crawlee crawlers are regular Python scripts, period. No launcher script required. You can instantiate a crawler in a FastAPI view handler for all I care

That's something I can see as a benefit mentioned in the list, that has more merit than "minimalistic and elegant".

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (vdusek) [reply README.md:192]: https://github.com/apify/crawlee-python/issues/776

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (vdusek) [reply README.md:190]: Updated.

File: README.md [hunk 3]
```diff
@@ -185,10 +185,12 @@ Why Crawlee is the preferred choice for web scraping and crawling?
...
+- **Headless browser support** – Crawlee natively supports the headless browser crawling via [Playwright](https://playwright.dev/python/).
```
[the remark this answers is not in the document]
Comment (vdusek) [reply README.md:191]: You're right. Removing it.

**Comment by vdusek** [comment]
@honzajavorek, thanks for your feedback.

File: README.md [hunk 4]
```diff
@@ -185,10 +185,12 @@ Why Crawlee is the preferred choice for web scraping and crawling?
 
 ### Why to use Crawlee rather than Scrapy?
 
-- Crawlee has out-of-the-box support for **headless browser** crawling (Playwright).
-- Crawlee has a **minimalistic & elegant interface** - Set up your scraper with fewer than 10 lines of code.
-- Complete **type hint** coverage.
-- Based on standard **Asyncio**.
+- **Asyncio-based** – Leveraging the standard [Asyncio](https://docs.python.org/3/library/asyncio.html) library, Crawlee delivers better performance and seamless compatibility with other modern asynchronous libraries.
```
[the remark this answers is not in the document]
Comment (honzajavorek) [reply README.md:188]: In text, they use asyncio, lowercase, no emphasis.

Screenshot 2024-12-03 at 13 43 14

Sometimes they use the monospace font. I'd personally use monospace font only when referring to the Python module in the context of documentation, and I'd use plain asyncio when referring to a "Python feature" or "Python's concept of asynchronous programming".

Screenshot 2024-12-03 at 13 44 12

**Comment by honzajavorek** [comment]
One more thing I didn't notice previously - I'm sorry! Most of the points start with "Crawlee something..." or "unlike Scrapy, Crawlee..." Given the heading already sets the scene, I think we can be shorter by just listing the benefits:

> - Newer project built with modern Python and complete type hint coverage for a better developer experience.
> - Its crawlers are regular Python scripts. You don't need a separate command to launch them, and you can integrate them directly into other applications.
> - Supports state persistence during interruptions, saving time and costs by avoiding the need to restart scraping pipelines from scratch after an issue.
> - Allows saving of multiple types of results in a single scraping run. Offers several storing options (see [datasets](https://crawlee.dev/python/api/class/Dataset) and [key-value stores](https://crawlee.dev/python/api/class/KeyValueStore)).

Something along these lines. Feel free to drop my suggestion or get loosely inspired by it.

**Review by janbuchar — APPROVED** [review]

**Pull request merged**